test/e2e/windows/memory_limits.go

```diff
@@ -0,0 +1,336 @@
+	calculatedNodeAlloc.Sub(nodeMem.hardEviction)
+
+	By(fmt.Sprintf("Checking stated allocatable memory %v against calculated allocatable memory %v", &nodeMem.allocatable, calculatedNodeAlloc))
+
+	// sanity check against stated allocatable
+	Expect(calculatedNodeAlloc.Cmp(nodeMem.allocatable)).To(Equal(0))
+}
+
+// Deploys `allocatablePods + 1` pods, each with a memory limit of `1/allocatablePods` of the total allocatable
+// memory, then confirms that the last pod failed because of failedScheduling
+func overrideAllocatableMemoryTest(f *framework.Framework, allocatablePods int) {
+	const (
+		podType = "memory_limit_test_pod"
+	)
+
+	totalAllocatable := getTotalAllocatableMemory(f)
+
+	memValue := totalAllocatable.Value()
+	memPerPod := memValue / int64(allocatablePods)
+	By(fmt.Sprintf("Deploying %d pods with mem limit %v, then one additional pod", allocatablePods, memPerPod))
+
+	// these should all work
+	pods := newMemLimitTestPods(allocatablePods, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
+	f.PodClient().CreateBatch(pods)
+
+	failurePods := newMemLimitTestPods(1, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
+	f.PodClient().Create(failurePods[0])
+
+	Eventually(func() bool {
+		eventList, err := f.ClientSet.CoreV1().Events(f.Namespace.Name).List(metav1.ListOptions{})
+		framework.ExpectNoError(err)
+		for _, e := range eventList.Items {
+			// Look for an event that shows FailedScheduling
+			if e.Type == "Warning" && e.Reason == "FailedScheduling" && e.InvolvedObject.Name == failurePods[0].ObjectMeta.Name {
+				framework.Logf("Found %+v event with message %+v", e.Reason, e.Message)
+				return true
+			}
+		}
+		return false
+	}, 3*time.Minute, 10*time.Second).Should(Equal(true))
+}
+
+// newMemLimitTestPods creates a list of pods (specification) for test.
+func newMemLimitTestPods(numPods int, imageName, podType string, memoryLimit string) []*v1.Pod {
+	var pods []*v1.Pod
+
+	memLimitQuantity, err := resource.ParseQuantity(memoryLimit)
+	framework.ExpectNoError(err)
+
+	for i := 0; i < numPods; i++ {
+
+		podName := "test-" + string(uuid.NewUUID())
+		pod := v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: podName,
+				Labels: map[string]string{
+					"type": podType,
+					"name": podName,
+				},
+			},
+			Spec: v1.PodSpec{
+				// Restart policy is always (default).
+				Containers: []v1.Container{
+					{
+						Image: imageName,
+						Name:  podName,
+						Resources: v1.ResourceRequirements{
+							Limits: v1.ResourceList{
+								v1.ResourceMemory: memLimitQuantity,
+							},
+						},
+					},
+				},
+				NodeSelector: map[string]string{
+					"beta.kubernetes.io/os": "windows",
+				},
+			},
+		}
+
+		pods = append(pods, &pod)
+	}
+
+	return pods
+}
+
+// getNodeMemory populates a nodeMemory struct with information from the first
+func getNodeMemory(f *framework.Framework) nodeMemory {
+	selector := labels.Set{"kubernetes.io/role": "agent"}.AsSelector()
```

Comment:
I don't think it's standard or universal to have the role "agent" on nodes

Reply:
Good to know, I'll switch the check to os=windows instead to align with the deployment